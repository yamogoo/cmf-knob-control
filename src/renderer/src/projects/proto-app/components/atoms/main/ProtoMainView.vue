<template lang="pug">
main(
  :class="[`${PROTO_APP_PREFIX}-main-view`]"
)
  slot
</template>

<script setup lang="ts">
import { PROTO_APP_PREFIX } from '@proto/config'
</script>

<style lang="scss">
$main-view: (
  sm: (),
  md: (),
  lg: ()
);

.#{$PROTO_APP_PREFIX}-main-view {
  position: relative;
  display: block;
  @include box(100%);
  background-color: none;
  overflow: hidden;
}
</style>
