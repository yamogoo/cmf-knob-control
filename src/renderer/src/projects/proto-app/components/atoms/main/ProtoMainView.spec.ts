import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import ProtoMainView from './ProtoMainView.vue'

describe('ProtoMainView', () => {
  test.each(['<p>Slot Content</p>'])('should render default slot (%s)', (content) => {
    const component = mount(ProtoMainView, {
      slots: {
        default: content
      }
    })

    const slotHtml = component.html()
    expect(slotHtml).toContain(content)
    expect(slotHtml).toMatchSnapshot()
  })
})
