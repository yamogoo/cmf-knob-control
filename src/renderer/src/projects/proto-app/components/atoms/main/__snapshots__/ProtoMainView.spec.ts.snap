// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProtoMainView > should render default slot (<p>Slot Content</p>) 1`] = `
"<main class="proto-app-main-view">
  <p>Slot Content</p>
</main>"
`;
